infra/Pos/Diffusion/Types.hs

```diff
@@ -77,15 +83,17 @@ dummyDiffusionLayer = DiffusionLayer
   where
     dummyDiffusion :: Applicative m => Diffusion m
     dummyDiffusion = Diffusion
-        { getBlocks          = \_ _ _ -> pure (Left (GetBlocksError "not implemented"))
+        { getBlocks          = \_ _ _ -> pure []
```

Comment:
Decision on 1: the result of `getBlocks` is a bunch of well-formed blocks (a well-formed block chain), order determined by the sender. They are not necessarily *valid* nor are they necessarily *in any order* nor is the chain itself necessarily *valid*. Logic layer can check that. Diffusion layer is responsible only for getting the well-formed data from the network here.

> returning [] is unacceptable.

I asked for the blocks, the peer said there were no blocks. What's the problem?

Reply:
Hm, on second thought, it's probably useful to keep the in-order requirement.

Reply:
>What's the problem?

It's not the problem, it's just a "where to perform some validation actions" question. It'd be convenient to use `getBlocks` if it guaranteed that the chain is linked and it's a valid response according to the protocol. Remember, there's a description of the protocol in `Pos.Block.Network.Types`. Some sanity checks like "i requested block for header `h` and got block with header `h`" would be nice to check somewhere, and I don't see any reason to do it outside of the diffusion layer. On the other hand, it's not that important where do we do them, because `getBlocks` is called once in the code, so even if this check is performed at the logic layer, we don't have code duplication. I'm good with any decision here, but I think contract should be documented in the `Diffusion` comments.